### PR TITLE
Size hint with delta is not negative

### PR DESCRIPTION
**Pull request by som-snytt**
Follow-up to https://github.com/scala/scala/pull/10642

### CODE REVIEW

File: src/library/scala/collection/mutable/Builder.scala [hunk 1]
```diff
@@ -54,9 +57,13 @@ trait Builder[-A, +To] extends Growable[A] { self =>
    *
    *  {{{
    *    if (coll.knownSize != -1)
-   *      sizeHint(coll.knownSize + delta)
+   *      if (coll.knownSize + delta < 0) sizeHint(0)
```
Comment (He-Pin) [inline src/library/scala/collection/mutable/Builder.scala]: `＜=`？

**Review by He-Pin — APPROVED** [review]
Lgtm

File: src/library/scala/collection/mutable/Builder.scala (hunk 1 above)
Comment (som-snytt) [reply src/library/scala/collection/mutable/Builder.scala]: it's pseudocode. 😄

File: src/library/scala/collection/mutable/Builder.scala [hunk 2]
```diff
@@ -40,6 +40,9 @@ trait Builder[-A, +To] extends Growable[A] { self =>
    *  The semantics of supplying a hint out of range, such as a size that is negative
    *  or unreasonably large, is not specified but is implementation-dependent.
    *
+   *  For example, the internal representation may be optimized for a size that is either
+   *  "as large as possible" or "as small as possible", or the hint may be ignored.
+   *
```
Comment (lrytz) [inline src/library/scala/collection/mutable/Builder.scala:39]: I still find these two paragraphs very confusing. What information are they adding to the existing comments?
  - "builder implementations are required to work correctly even if the hint is wrong"
  - "the default implementation simply ignores the hint"

Comment (som-snytt) [reply src/library/scala/collection/mutable/Builder.scala:39]: I "think" I was trying address the "willies" noted at item 3 here:
https://users.scala-lang.org/t/migrating-2-12-2-13-problems-with-arraybuffer-immutable-seq-missing-capabilities/9700/10
I'll delete the paragraphs as "trying to explain too much". "But what if the hint is _way_ wrong?" "It must work even if the hint is way wrong." That could be a FAQ...

Comment (som-snytt) [reply src/library/scala/collection/mutable/Builder.scala:39]: OK, I accommodated the "willies" by changing `i.e.` to `e.g.` and adding the other way a hint can be wrong.

Oh, that means the builder is not allowed to throw on a bad hint.

File: src/library/scala/collection/mutable/Builder.scala (hunk 1 above)
Comment (som-snytt) [reply src/library/scala/collection/mutable/Builder.scala]: oh I meant to fix this, so long as I had to edit it again.

I hope it cancels the C.I. run.

**Review by jxnu-liguobin — APPROVED** [review]
lgtm

**Review by lrytz — APPROVED** [review]
Thank you 🙏

**Pull request merged**